src/components/ui/time-slider/time-slider.tsx

```diff
@@ -0,0 +1,141 @@
+import { observer } from "mobx-react-lite";
+import React, { useEffect, useState } from "react";
+import MapSlider from "../../../assets/icons/map-slider.svg";
+import PauseIcon from "../../../assets/icons/pause-icon.svg";
+import PlayIcon from "../../../assets/icons/play-icon.svg";
+import TimeSliderThumb from "../../../assets/icons/time-slider-thumb.svg";
+import { formatDateString } from "../../../utilities/date-utils";
+import { UIButton } from "../ui-button";
+import { UIButtonContainer } from "../ui-button-container";
+import { DateRangeSliderThumb } from "./date-range-slider-thumb";
+import { SliderThumb } from "./slider-thumb";
+import { TimeLine } from "./time-line";
+import {
+  labelHeight, labelOffsets, mapSliderThumbOffset, timeLineTop, timeSliderThumbOffset
+} from "./time-slider-contsants";
+import "./time-slider.scss";
+
+const labelBaseTop = timeLineTop - labelHeight / 2;
+let lastFrameTime = Date.now();
+const animationRate = 0.1;
+let animationTimeout = 0;
+
+interface ITimeSliderProps {
+  dateMax: number;
+  dateMin: number;
+}
+export const TimeSlider = observer(function TimeSlider({ dateMax, dateMin }: ITimeSliderProps) {
+  const [upperRangePercent, setUpperRangePercent] = useState(1);
+  const [lowerRangePercent, setLowerRangePercent] = useState(0);
+  const [mapPercent, setMapPercent] = useState(0);
+  const [timePercent, setTimePercent] = useState(1);
+  const [animating, setAnimating] = useState(false);
+
+  useEffect(() => {
+    setMapPercent(value => Math.min(upperRangePercent, Math.max(lowerRangePercent, value)));
+    setTimePercent(value => Math.min(upperRangePercent, Math.max(lowerRangePercent, value)));
+  }, [upperRangePercent, lowerRangePercent]);
+
+  useEffect(() => {
+    if (animating && !animationTimeout) {
+      animationTimeout = setTimeout(() => {
+        animationTimeout = 0;
```

Comment:
Setting the `animationTimeout` to 0 here seems strange. Since there is an `animating` state variable it shouldn't be necessary to check and set the animationTimeout too. But I'm guessing using this fixes something, so a comment about why it is needed would be useful.

Reply:
Yes, because this is in an effect with `upperRangePercent` as a dependency, it's possible to set extra timeouts by moving the upper thumb during animation. It's a weird edge case that might not be worth addressing, but hopefully it will be fixed when I'm able to use mobx observable state instead of react state.